firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionLifecycleService.kt

```diff
@@ -196,6 +228,16 @@ internal class SessionLifecycleService() : Service() {
         sessionId.let { datastore.updateSessionId(it) }
       }
     }
+
+    /** Calculate whether we should sample events using [SessionsSettings] data. */
+    private fun shouldCollectEvents(): Boolean {
+      // Sampling rate of 1 means the SDK will send every event.
+      return randomValueForSampling <= SessionsSettings.instance.samplingRate
```

Comment:
I don't think there's any need for this sampling lever since we have the enablement flag. What does it mean to sample a session? Does everything still get associated with the previous session? What does it mean for metrics? I believe we just added this to settings to give us the ability to shut things down remotely if it was too much traffic, so IMO we should either remove it altogether or move this sampling logic to the firelog uploader to just prevent upload.

Reply:
Discussed in chat